src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -158,54 +175,88 @@ private boolean executeHTTPServiceAndCheckStatusCode() throws IOException {
+
+        CredentialsProvider credsProvider = new BasicCredentialsProvider();
+
+        if (proxyUser != null && proxyPassword != null) {
+          credsProvider.setCredentials(
+            new AuthScope(proxyHost, proxyPort),
+            new UsernamePasswordCredentials(proxyUser, proxyPassword));
+        }
+
+        HttpHost proxy = new HttpHost(proxyHost, proxyPort);
+        httpClient = HttpClients.custom()
+          .setDefaultCredentialsProvider(credsProvider)
+          .setProxy(proxy)
+          .build();
+      }
+
+      if (url.getProtocol().equalsIgnoreCase("https")) {
+        // Disable SSLv3
         System.setProperty("https.protocols", "TLSv1,TLSv1.1,TLSv1.2");
         if (config.getDisableSSLValidation()) {
           disableSSLValidation();
         }
       }
-      conn.setRequestMethod(config.getMethod().toUpperCase());
-      conn.setConnectTimeout(config.getConnectTimeout());
-      conn.setReadTimeout(config.getReadTimeout());
-      conn.setInstanceFollowRedirects(config.getFollowRedirects());
-      conn.addRequestProperty("charset", config.getCharset());
-      for (Map.Entry<String, String> propertyEntry : headers.entrySet()) {
-        conn.addRequestProperty(propertyEntry.getKey(), propertyEntry.getValue());
-      }
-      //Default contentType value would be added in the request properties if user has not added in the headers.
+
       if (config.getMethod().equals("POST") || config.getMethod().equals("PUT")) {
         if (!headers.containsKey("Content-Type")) {
-          conn.addRequestProperty("Content-Type", contentType);
+          headers.put("Content-Type", contentType);
         }
       }
+
       if (!messageBuffer.isEmpty()) {
-        conn.setDoOutput(true);
-        try (OutputStream outputStream = conn.getOutputStream()) {
-          outputStream.write(messageBuffer.getMessage().trim().getBytes(config.getCharset()));
+        String requestBodyString = messageBuffer.getMessage();
+        if (requestBodyString != null) {
+          StringEntity requestBody = new StringEntity(requestBodyString, Charsets.UTF_8.toString());
+          request.setEntity(requestBody);
         }
       }
-      httpStatusCode = conn.getResponseCode();
+
+      for (Map.Entry<String, String> propertyEntry : headers.entrySet()) {
+        request.addHeader(propertyEntry.getKey(), propertyEntry.getValue());
+      }
+
+      response = httpClient.execute(request);
+
+      httpStatusCode = response.getStatusLine().getStatusCode();
+      LOG.debug("Response HTTP Status code: {}", httpStatusCode);
+
     } catch (MalformedURLException | ProtocolException e) {
       throw new IllegalStateException("Error opening url connection. Reason: " + e.getMessage(), e);
     } catch (IOException e) {
-      LOG.warn("Error making {} request to url {} with headers {}.", config.getMethod(), config.getUrl(), headers);
+      LOG.warn("Error making {} request to url {}.", config.getMethod(), config.getUrl());
```

Comment:
We should not log the header informations in the logs.

Reply:
Headers are removed in this change, previously it was there